### DISCUSSION TITLE
Want to remove ads on static pages

### DISCUSSION THREAD

**Discussion by yudiz-nikunjt** [vercel/next.js#40294] [Help]
I have used server side rendering on home page and contact us static page. I showed an ad poster on both page. but i only want to show that ad to home page. I have put a condition there so if i directly go to contact page the ad poster will not show. It works perfectly in this scenario. But if i will go to Contact page from Home page link through client side routing, The ad poster is still there, Its not removed because ad is already added through server side so If i will go to Contact page through client side routing it will not work.

Anyone have any idea How can I resolve this issue?

**Comment by icyJoseph** [comment]
Could you try to reproduce this on https://stackblitz.com/? 

If the Ad is just a React component, you should be able to simply add it to the Home Page JSX and the SPA navigation should take care of removing it when you navigate to contact, with one caveat, that the Ad doesn't mutate the DOM without React's knowledge. 

For example if the Ad components loads a script, or if it does direct DOM manipulation, then the SPA transitions won't be able to remove it, under certain conditions. This is why I would like to see a reproduction of your use case.

**Reply by yudiz-nikunjt** [reply]
I have rendered Ads component to the _app.js page. so first time it will show to all the pages.

**Reply by icyJoseph** [reply]
Why not keep it for home only? Or perhaps use [Per-Page layouts](https://nextjs.org/docs/basic-features/layouts#per-page-layouts), so that the default case is no ads, and the home page has a layout that includes Ads?

**Reply by yudiz-nikunjt** [reply]
Actually Project is big. there are several dynamic and static pages. So i have added that ad component to the _app.js.

**Reply by icyJoseph** [reply]
Mmm then you could get away with it by using the router, `useRouter`, to see if, the router is ready and if the pathname matches the home page. If both are true render the Ad. Maybe you need to do some window checks to prevent hydration errors because of divergent HTML between client and server.

https://nextjs.org/docs/api-reference/next/router#router-object

A bit of a hack but it should do the trick. Just place this logic inside the Ad rendering component, to keep it all clean.

**Reply by yudiz-nikunjt** [reply]
Hmm I have tried it but it doesn't worked. I destroyed ads on static page. Issue resolved

**Reply by yudiz-nikunjt** [reply]
Thanks
